### ISSUE TITLE
Change `shorthand`/`type` algo to treat functions as `Literal`

### ISSUE THREAD

**Issue by kredati** [kredati/ludus-prelude#20]
As I work out how `match` works, and now that I've added an `Enum` type, it seems to me that one of the more common use cases is going to be to use an `Enum` to model the API of a module. (Exactly how modules work TBD.)

Consider:
```javascript
const Model = Struct({count: Number})

const inc = Num.add(1, __)
const dec = Num.sub(__, 1)

const Action = Enum({inc, dec})

const update = defn({name: 'update', sign: [Model, Action, Model],
    body: match(
        {value: action, as: Action, with: (state, action) => action},
        [inc, ({count}) => Model.create({count: inc(count)})],
        [dec, ({count}) => Model.create({count: dec(count)})])
```
The `match` ergonomics here are a draft, and this still feels like a bit of a kludge. **But**, you can see here that you want to switch on the `Enum` members, and you want `Enum` members to be functions, and you therefore want the guards here to be, e.g. `Literal(inc)`, without having to be explicit about it.

---
So that's the rationale. The change to the algo should be trivial to make.

**Comment by kredati** [comment]
Obviously, the body of `update` above would be `({count}, action) => Model.create({count: action(count)})`. But for more complex use cases (with one part of the Model updating in one way with one function, and another way in another function), this is desirable behavior. (Even, I should say, with a different (probably better) model here, which would also include a payload value of some kind.)

**Comment by kredati** [comment]
Solved as part of #21
